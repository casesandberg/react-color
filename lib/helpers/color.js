'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _tinycolor = require('../../modules/tinycolor2');

var _tinycolor2 = _interopRequireDefault(_tinycolor);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = {

  simpleCheckForValidColor: function simpleCheckForValidColor(data) {
    var keysToCheck = ['r', 'g', 'b', 'a', 'h', 's', 'a', 'v'];
    var checked = 0;
    var passed = 0;
    for (var i = 0; i < keysToCheck.length; i++) {
      var letter = keysToCheck[i];
      if (data[letter]) {
        checked++;
        if (!isNaN(data[letter])) {
          passed++;
        }
      }
    }

    if (checked === passed) {
      return data;
    }
  },

  toState: function toState(data, oldHue) {
    var color = data.hex ? (0, _tinycolor2.default)(data.hex) : (0, _tinycolor2.default)(data);
    var hsl = color.toHsl();
    var hsv = color.toHsv();
    if (hsl.s === 0) {
      hsl.h = oldHue || 0;
      hsv.h = oldHue || 0;
    }

    return {
      hsl: hsl,
      hex: '#' + color.toHex(),
      rgb: color.toRgb(),
      hsv: hsv,
      oldHue: data.h || oldHue || hsl.h,
      source: data.source
    };
  },

  isValidHex: function isValidHex(hex) {
    return (0, _tinycolor2.default)(hex).isValid();
  }

};