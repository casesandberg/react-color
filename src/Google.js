export default from './components/google/Google'
