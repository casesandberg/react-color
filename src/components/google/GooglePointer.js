import React from 'react'
import reactCSS from 'reactcss'
import PropTypes from 'prop-types'

export const GooglePointer = (props) => {
  const styles = reactCSS({
    'default': {
      picker: {
        width: '20px',
        height: '20px',
        borderRadius: '22px',
        transform: 'translate(-10px, -7px)',
        background: `hsl(${ Math.round(props.hsl.h) }, 100%, 50%)`,
        border: '2px white solid',
      },
    },
  })

  return (
    <div style={ styles.picker } />
  )
}

GooglePointer.propTypes = {
  hsl: PropTypes.shape({
    h: PropTypes.number,
    s: PropTypes.number,
    l: PropTypes.number,
    a: PropTypes.number,
  }),
}

GooglePointer.defaultProps = {
  hsl: { a: 1, h: 249.94, l: 0.2, s: 0.50 },
}

export default GooglePointer
