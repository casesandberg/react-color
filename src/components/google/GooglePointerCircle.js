import React from 'react'
import reactCSS from 'reactcss'
import PropTypes from 'prop-types'

export const GooglePointerCircle = (props) => {
  const styles = reactCSS({
    'default': {
      picker: {
        width: '20px',
        height: '20px',
        borderRadius: '22px',
        border: '2px #fff solid',
        transform: 'translate(-12px, -13px)',
        background: `hsl(${ Math.round(props.hsl.h) }, ${ Math.round(props.hsl.s * 100 ) }%, ${ Math.round(props.hsl.l * 100) }%)`,
      },
    },
  })

  return (
    <div style={ styles.picker } />
  )
}

GooglePointerCircle.propTypes = {
  hsl: PropTypes.shape({
    h: PropTypes.number,
    s: PropTypes.number,
    l: PropTypes.number,
    a: PropTypes.number,
  }),
}

GooglePointerCircle.defaultProps = {
  hsl: { a: 1, h: 249.94, l: 0.2, s: 0.50 },
}

export default GooglePointerCircle