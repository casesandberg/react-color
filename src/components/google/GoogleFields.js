import React from 'react'
import reactCSS from 'reactcss'
import color from '../../helpers/color'
import { EditableInput } from '../common'


export const GoogleFields = ({ onChange, rgb, hsl, hex, hsv }) => {

  const handleChange = (data, e) => {
    if (data.hex) {      
      color.isValidHex(data.hex) && onChange({
        hex: data.hex,
        source: 'hex',
      }, e)
    } else if (data.rgb) {
      const values = data.rgb.split(',')
      color.isvalidColorString(data.rgb, 'rgb') && onChange({
        r: values[0],
        g: values[1],
        b: values[2],
        a: 1,
        source: 'rgb',
      }, e)
    } else if (data.hsv){
      const values = data.hsv.split(',')      
      if (color.isvalidColorString(data.hsv, 'hsv')){
        values[2] = values[2].replace('%', '')
        values[1] = values[1].replace('%', '')
        values[0] = values[0].replace('째', '')
        if (values[1] == 1) {
          values[1] = 0.01
        } else if (values[2] == 1) {
          values[2] = 0.01
        }
        onChange({
          h: Number(values[0]),
          s: Number(values[1]),
          v: Number(values[2]),
          source: 'hsv',
        }, e)
      }
    } else if (data.hsl) {
      const values = data.hsl.split(',')
      if (color.isvalidColorString(data.hsl, 'hsl')){
        values[2] = values[2].replace('%', '')
        values[1] = values[1].replace('%', '')
        values[0] = values[0].replace('째', '')
        if (hsvValue[1] == 1) {
          hsvValue[1] = 0.01
        } else if (hsvValue[2] == 1) {
          hsvValue[2] = 0.01
        }
        onChange({
          h: Number(values[0]),
          s: Number(values[1]),
          v: Number(values[2]),
          source: 'hsl',
        }, e)
      }
    }
  }

  const styles = reactCSS({
    'default': {
      wrap: {
        display: 'flex',
        height: '100px',
        marginTop: '4px',
      },
      fields: {
        width: '100%',
      },
      column: {
        paddingTop: '10px',
        display: 'flex',
        justifyContent: 'space-between',
      },
      double: {
        padding: '0px 4.4px',
        boxSizing: 'border-box',
      },
      input: {
        width: '100%',
        height: '38px',
        boxSizing: 'border-box',
        padding: '4px 10% 3px',
        textAlign: 'center',
        border: '1px solid #dadce0',
        fontSize: '11px',
        textTransform: 'lowercase',
        borderRadius: '5px',
        outline: 'none',
        fontFamily: 'Roboto,Arial,sans-serif',
      },
      input2: {
        height: '38px',
        width: '100%',
        border: '1px solid #dadce0',
        boxSizing: 'border-box',
        fontSize: '11px',
        textTransform: 'lowercase',
        borderRadius: '5px',
        outline: 'none',
        paddingLeft: '10px',
        fontFamily: 'Roboto,Arial,sans-serif',
      },
      label: {
        textAlign: 'center',
        fontSize: '12px',
        background: '#fff',
        position: 'absolute',
        textTransform: 'uppercase',
        color: '#3c4043',
        width: '35px',
        top: '-6px',
        left: '0',
        right: '0',
        marginLeft: 'auto',
        marginRight: 'auto',
        fontFamily: 'Roboto,Arial,sans-serif',
      },
      label2: {
        left: '10px',
        textAlign: 'center',
        fontSize: '12px',
        background: '#fff',
        position: 'absolute',
        textTransform: 'uppercase',
        color: '#3c4043',
        width: '32px',
        top: '-6px',
        fontFamily: 'Roboto,Arial,sans-serif',
      },
      single: {
        flexGrow: '1',
        margin: '0px 4.4px',
      },
    },
  })

  const rgbValue = `${ rgb.r }, ${ rgb.g }, ${ rgb.b }`    
  const hslValue = `${ Math.round(hsl.h) }째, ${ Math.round(hsl.s * 100) }%, ${ Math.round(hsl.l * 100) }%`
  const hsvValue = `${ Math.round(hsv.h) }째, ${ Math.round(hsv.s * 100) }%, ${ Math.round(hsv.v * 100) }%`

  return (
    <div style={ styles.wrap } className="flexbox-fix">
      <div style={ styles.fields }>
        <div style={ styles.double }>
          <EditableInput
            style={{ input: styles.input, label: styles.label }}
            label="hex"
            value={ hex }
            onChange={ handleChange }
          />
        </div>
        <div style={ styles.column }>
          <div style={ styles.single }>
            <EditableInput
              style={{ input: styles.input2, label: styles.label2 }}
              label="rgb"
              value={ rgbValue }
              onChange={ handleChange }  
            />
          </div>
          <div style={ styles.single }>
            <EditableInput
              style={{ input: styles.input2, label: styles.label2 }}
              label="hsv"
              value={ hsvValue }
              onChange={ handleChange }
            />
          </div>
          <div style={ styles.single }>
            <EditableInput
              style={{ input: styles.input2, label: styles.label2 }}
              label="hsl"
              value={ hslValue }
              onChange={ handleChange }
            />
          </div>
        </div>
      </div>
    </div>
  )
}

export default GoogleFields