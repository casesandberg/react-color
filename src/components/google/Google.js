import React from 'react'
import PropTypes from 'prop-types'
import reactCSS from 'reactcss'
import merge from 'lodash/merge'

import { ColorWrap, Saturation, Hue } from '../common'
import GooglePointerCircle from './GooglePointerCircle'
import GooglePointer from './GooglePointer'
import GoogleFields from './GoogleFields'

export const Google = ({ width, onChange, rgb, hsl, hsv, hex, header,
  styles: passedStyles = {}, className = '' }) => {
  const styles = reactCSS(merge({
    'default': {
      picker: {
        width,
        background: '#fff',
        border: '1px solid #dfe1e5',
        boxSizing: 'initial',
        display: 'flex',
        flexWrap: 'wrap',
        borderRadius: '8px 8px 0px 0px',
      },
      head: {
        height: '57px',
        width: '100%',
        paddingTop: '16px',
        paddingBottom: '16px',
        paddingLeft: '16px',
        fontSize: '20px',
        boxSizing: 'border-box',
        fontFamily: 'Roboto-Regular,HelveticaNeue,Arial,sans-serif',
      },
      saturation: {
        width: '70%',
        padding: '0px',
        position: 'relative',
        overflow: 'hidden',
      },
      swatch: {
        width: '30%',
        height: '228px',
        padding: '0px',
        background: `rgba(${ rgb.r }, ${ rgb.g }, ${ rgb.b }, 1)`,
        position: 'relative',
        overflow: 'hidden',
      },
      body: {
        margin: 'auto',
        width: '95%',
      },
      controls: {
        display: 'flex',
        boxSizing: 'border-box',
        height: '52px',
        paddingTop: '22px',
      },
      color: {
        width: '32px',
      },
      hue: {
        height: '8px',
        position: 'relative',
        margin: '0px 16px 0px 16px',
        width: '100%',
      },
      Hue: {
        radius: '2px',
      },
    },
  }, passedStyles))
  return (
    <div style={ styles.picker } className={ `google-picker ${ className }` }>
      <div style={ styles.head }>{ header }</div>
      <div style={ styles.swatch } />
      <div style={ styles.saturation }>
        <Saturation
          hsl={ hsl }
          hsv={ hsv }
          pointer={ GooglePointerCircle }
          onChange={ onChange }
        />
      </div>
      <div style={ styles.body }>
        <div style={ styles.controls } className="flexbox-fix">
          <div style={ styles.hue }>
            <Hue
              style={ styles.Hue }
              hsl={ hsl }
              radius="4px"
              pointer={ GooglePointer }
              onChange={ onChange }
            />
          </div>
        </div>
        <GoogleFields
          rgb={ rgb }
          hsl={ hsl }
          hex={ hex }
          hsv={ hsv }
          onChange={ onChange }
        />
      </div>
    </div>
  )
}

Google.propTypes = {
  width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  styles: PropTypes.object,
  header: PropTypes.string,

}

Google.defaultProps = {
  width: 652,
  styles: {},
  header: 'Color picker',
}

export default ColorWrap(Google)