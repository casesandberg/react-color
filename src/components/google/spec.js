import React from 'react'
import renderer from 'react-test-renderer'
import color, { red } from '../../helpers/color'
import { mount } from 'enzyme'

import Google from './Google'
import GoogleFields from './GoogleFields'
import GooglePointer from './GooglePointer'
import GooglePointerCircle from './GooglePointerCircle'


test('Google renders correctly', () => {
  const tree = renderer.create(<Google { ...red } />).toJSON()
  expect(tree).toMatchSnapshot()
})
test('Google onChange events correctly', () => {
  const changeSpy = jest.fn((data) => {
    expect(color.simpleCheckForValidColor(data)).toBeTruthy()
  })
  const tree = mount(
    <Google { ...red } onChange={ changeSpy } />,
  )
  expect(changeSpy).toHaveBeenCalledTimes(0)
})


test('GoogleFields renders correctly', () => {
  const tree = renderer.create(
    <GoogleFields { ...red } />,
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('GooglePointer renders correctly', () => {
  const tree = renderer.create(
    <GooglePointer />,
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('GooglePointerCircle renders correctly', () => {
  const tree = renderer.create(
    <GooglePointerCircle />,
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Google renders custom styles correctly', () => {
  const tree = renderer.create(
    <Google styles={{ default: { picker: { width: '200px' } } }} />,
  ).toJSON()
  expect(tree.props.style.width).toBe('200px')
})

test('Google renders correctly with width', () => {
  const tree = renderer.create(
    <Google width={ 200 } />,
  ).toJSON()
  expect(tree.props.style.width).toBe(200)
})

test('Google custom header correctly', () => {
  const tree = mount(
    <Google header="Change the color!!!" />,
  )
  expect(tree.instance().props.header).toBe('Change the color!!!')
})
