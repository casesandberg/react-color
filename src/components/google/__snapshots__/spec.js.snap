// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Google renders correctly 1`] = `
<div
  className="google-picker "
  style={
    Object {
      "MozBorderRadius": "8px 8px 0px 0px",
      "OBorderRadius": "8px 8px 0px 0px",
      "WebkitBorderRadius": "8px 8px 0px 0px",
      "background": "#fff",
      "border": "1px solid #dfe1e5",
      "borderRadius": "8px 8px 0px 0px",
      "boxSizing": "initial",
      "display": "flex",
      "flexWrap": "wrap",
      "msBorderRadius": "8px 8px 0px 0px",
      "width": 652,
    }
  }
>
  <div
    style={
      Object {
        "boxSizing": "border-box",
        "fontFamily": "Roboto-Regular,HelveticaNeue,Arial,sans-serif",
        "fontSize": "20px",
        "height": "57px",
        "paddingBottom": "16px",
        "paddingLeft": "16px",
        "paddingTop": "16px",
        "width": "100%",
      }
    }
  >
    Color picker
  </div>
  <div
    style={
      Object {
        "background": "rgba(34, 25, 77, 1)",
        "height": "228px",
        "overflow": "hidden",
        "padding": "0px",
        "position": "relative",
        "width": "30%",
      }
    }
  />
  <div
    style={
      Object {
        "overflow": "hidden",
        "padding": "0px",
        "position": "relative",
        "width": "70%",
      }
    }
  >
    <div
      onMouseDown={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "MozBorderRadius": undefined,
          "OBorderRadius": undefined,
          "WebkitBorderRadius": undefined,
          "background": "hsl(249.99999999999994,100%, 50%)",
          "borderRadius": undefined,
          "bottom": "0px",
          "left": "0px",
          "msBorderRadius": undefined,
          "position": "absolute",
          "right": "0px",
          "top": "0px",
        }
      }
    >
      <style>
        
                  .saturation-white {
                    background: -webkit-linear-gradient(to right, #fff, rgba(255,255,255,0));
                    background: linear-gradient(to right, #fff, rgba(255,255,255,0));
                  }
                  .saturation-black {
                    background: -webkit-linear-gradient(to top, #000, rgba(0,0,0,0));
                    background: linear-gradient(to top, #000, rgba(0,0,0,0));
                  }
                
      </style>
      <div
        className="saturation-white"
        style={
          Object {
            "MozBorderRadius": undefined,
            "OBorderRadius": undefined,
            "WebkitBorderRadius": undefined,
            "borderRadius": undefined,
            "bottom": "0px",
            "left": "0px",
            "msBorderRadius": undefined,
            "position": "absolute",
            "right": "0px",
            "top": "0px",
          }
        }
      >
        <div
          className="saturation-black"
          style={
            Object {
              "MozBorderRadius": undefined,
              "MozBoxShadow": undefined,
              "OBorderRadius": undefined,
              "OBoxShadow": undefined,
              "WebkitBorderRadius": undefined,
              "WebkitBoxShadow": undefined,
              "borderRadius": undefined,
              "bottom": "0px",
              "boxShadow": undefined,
              "left": "0px",
              "msBorderRadius": undefined,
              "msBoxShadow": undefined,
              "position": "absolute",
              "right": "0px",
              "top": "0px",
            }
          }
        />
        <div
          style={
            Object {
              "cursor": "default",
              "left": "66.66666666666667%",
              "position": "absolute",
              "top": "70%",
            }
          }
        >
          <div
            style={
              Object {
                "MozBorderRadius": "22px",
                "MozTransform": "translate(-12px, -13px)",
                "OBorderRadius": "22px",
                "OTransform": "translate(-12px, -13px)",
                "WebkitBorderRadius": "22px",
                "WebkitTransform": "translate(-12px, -13px)",
                "background": "hsl(250, 50%, 20%)",
                "border": "2px #fff solid",
                "borderRadius": "22px",
                "height": "20px",
                "msBorderRadius": "22px",
                "msTransform": "translate(-12px, -13px)",
                "transform": "translate(-12px, -13px)",
                "width": "20px",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "margin": "auto",
        "width": "95%",
      }
    }
  >
    <div
      className="flexbox-fix"
      style={
        Object {
          "boxSizing": "border-box",
          "display": "flex",
          "height": "52px",
          "paddingTop": "22px",
        }
      }
    >
      <div
        style={
          Object {
            "height": "8px",
            "margin": "0px 16px 0px 16px",
            "position": "relative",
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "MozBorderRadius": "4px",
              "MozBoxShadow": undefined,
              "OBorderRadius": "4px",
              "OBoxShadow": undefined,
              "WebkitBorderRadius": "4px",
              "WebkitBoxShadow": undefined,
              "borderRadius": "4px",
              "bottom": "0px",
              "boxShadow": undefined,
              "left": "0px",
              "msBorderRadius": "4px",
              "msBoxShadow": undefined,
              "position": "absolute",
              "right": "0px",
              "top": "0px",
            }
          }
        >
          <div
            className="hue-horizontal"
            onMouseDown={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "MozBorderRadius": "4px",
                "OBorderRadius": "4px",
                "WebkitBorderRadius": "4px",
                "borderRadius": "4px",
                "height": "100%",
                "msBorderRadius": "4px",
                "padding": "0 2px",
                "position": "relative",
              }
            }
          >
            <style>
              
                          .hue-horizontal {
                            background: linear-gradient(to right, #f00 0%, #ff0 17%, #0f0
                              33%, #0ff 50%, #00f 67%, #f0f 83%, #f00 100%);
                            background: -webkit-linear-gradient(to right, #f00 0%, #ff0
                              17%, #0f0 33%, #0ff 50%, #00f 67%, #f0f 83%, #f00 100%);
                          }
              
                          .hue-vertical {
                            background: linear-gradient(to top, #f00 0%, #ff0 17%, #0f0 33%,
                              #0ff 50%, #00f 67%, #f0f 83%, #f00 100%);
                            background: -webkit-linear-gradient(to top, #f00 0%, #ff0 17%,
                              #0f0 33%, #0ff 50%, #00f 67%, #f0f 83%, #f00 100%);
                          }
                        
            </style>
            <div
              style={
                Object {
                  "left": "69.44444444444443%",
                  "position": "absolute",
                }
              }
            >
              <div
                style={
                  Object {
                    "MozBorderRadius": "22px",
                    "MozTransform": "translate(-10px, -7px)",
                    "OBorderRadius": "22px",
                    "OTransform": "translate(-10px, -7px)",
                    "WebkitBorderRadius": "22px",
                    "WebkitTransform": "translate(-10px, -7px)",
                    "background": "hsl(250, 100%, 50%)",
                    "border": "2px white solid",
                    "borderRadius": "22px",
                    "height": "20px",
                    "msBorderRadius": "22px",
                    "msTransform": "translate(-10px, -7px)",
                    "transform": "translate(-10px, -7px)",
                    "width": "20px",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="flexbox-fix"
      style={
        Object {
          "display": "flex",
          "height": "100px",
          "marginTop": "4px",
        }
      }
    >
      <div
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "padding": "0px 4.4px",
            }
          }
        >
          <div
            style={
              Object {
                "position": "relative",
              }
            }
          >
            <input
              id="rc-editable-input-1"
              onBlur={[Function]}
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder={undefined}
              spellCheck="false"
              style={
                Object {
                  "MozBorderRadius": "5px",
                  "OBorderRadius": "5px",
                  "WebkitBorderRadius": "5px",
                  "border": "1px solid #dadce0",
                  "borderRadius": "5px",
                  "boxSizing": "border-box",
                  "fontFamily": "Roboto,Arial,sans-serif",
                  "fontSize": "11px",
                  "height": "38px",
                  "msBorderRadius": "5px",
                  "outline": "none",
                  "padding": "4px 10% 3px",
                  "textAlign": "center",
                  "textTransform": "lowercase",
                  "width": "100%",
                }
              }
              value="#22194D"
            />
            <label
              htmlFor="rc-editable-input-1"
              onMouseDown={[Function]}
              style={
                Object {
                  "background": "#fff",
                  "color": "#3c4043",
                  "fontFamily": "Roboto,Arial,sans-serif",
                  "fontSize": "12px",
                  "left": "0",
                  "marginLeft": "auto",
                  "marginRight": "auto",
                  "position": "absolute",
                  "right": "0",
                  "textAlign": "center",
                  "textTransform": "uppercase",
                  "top": "-6px",
                  "width": "35px",
                }
              }
            >
              hex
            </label>
          </div>
        </div>
        <div
          style={
            Object {
              "WebkitJustifyContent": "space-between",
              "display": "flex",
              "justifyContent": "space-between",
              "paddingTop": "10px",
            }
          }
        >
          <div
            style={
              Object {
                "flexGrow": "1",
                "margin": "0px 4.4px",
              }
            }
          >
            <div
              style={
                Object {
                  "position": "relative",
                }
              }
            >
              <input
                id="rc-editable-input-2"
                onBlur={[Function]}
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder={undefined}
                spellCheck="false"
                style={
                  Object {
                    "MozBorderRadius": "5px",
                    "OBorderRadius": "5px",
                    "WebkitBorderRadius": "5px",
                    "border": "1px solid #dadce0",
                    "borderRadius": "5px",
                    "boxSizing": "border-box",
                    "fontFamily": "Roboto,Arial,sans-serif",
                    "fontSize": "11px",
                    "height": "38px",
                    "msBorderRadius": "5px",
                    "outline": "none",
                    "paddingLeft": "10px",
                    "textTransform": "lowercase",
                    "width": "100%",
                  }
                }
                value="34, 25, 77"
              />
              <label
                htmlFor="rc-editable-input-2"
                onMouseDown={[Function]}
                style={
                  Object {
                    "background": "#fff",
                    "color": "#3c4043",
                    "fontFamily": "Roboto,Arial,sans-serif",
                    "fontSize": "12px",
                    "left": "10px",
                    "position": "absolute",
                    "textAlign": "center",
                    "textTransform": "uppercase",
                    "top": "-6px",
                    "width": "32px",
                  }
                }
              >
                rgb
              </label>
            </div>
          </div>
          <div
            style={
              Object {
                "flexGrow": "1",
                "margin": "0px 4.4px",
              }
            }
          >
            <div
              style={
                Object {
                  "position": "relative",
                }
              }
            >
              <input
                id="rc-editable-input-3"
                onBlur={[Function]}
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder={undefined}
                spellCheck="false"
                style={
                  Object {
                    "MozBorderRadius": "5px",
                    "OBorderRadius": "5px",
                    "WebkitBorderRadius": "5px",
                    "border": "1px solid #dadce0",
                    "borderRadius": "5px",
                    "boxSizing": "border-box",
                    "fontFamily": "Roboto,Arial,sans-serif",
                    "fontSize": "11px",
                    "height": "38px",
                    "msBorderRadius": "5px",
                    "outline": "none",
                    "paddingLeft": "10px",
                    "textTransform": "lowercase",
                    "width": "100%",
                  }
                }
                value="250°, 67%, 30%"
              />
              <label
                htmlFor="rc-editable-input-3"
                onMouseDown={[Function]}
                style={
                  Object {
                    "background": "#fff",
                    "color": "#3c4043",
                    "fontFamily": "Roboto,Arial,sans-serif",
                    "fontSize": "12px",
                    "left": "10px",
                    "position": "absolute",
                    "textAlign": "center",
                    "textTransform": "uppercase",
                    "top": "-6px",
                    "width": "32px",
                  }
                }
              >
                hsv
              </label>
            </div>
          </div>
          <div
            style={
              Object {
                "flexGrow": "1",
                "margin": "0px 4.4px",
              }
            }
          >
            <div
              style={
                Object {
                  "position": "relative",
                }
              }
            >
              <input
                id="rc-editable-input-4"
                onBlur={[Function]}
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder={undefined}
                spellCheck="false"
                style={
                  Object {
                    "MozBorderRadius": "5px",
                    "OBorderRadius": "5px",
                    "WebkitBorderRadius": "5px",
                    "border": "1px solid #dadce0",
                    "borderRadius": "5px",
                    "boxSizing": "border-box",
                    "fontFamily": "Roboto,Arial,sans-serif",
                    "fontSize": "11px",
                    "height": "38px",
                    "msBorderRadius": "5px",
                    "outline": "none",
                    "paddingLeft": "10px",
                    "textTransform": "lowercase",
                    "width": "100%",
                  }
                }
                value="250°, 50%, 20%"
              />
              <label
                htmlFor="rc-editable-input-4"
                onMouseDown={[Function]}
                style={
                  Object {
                    "background": "#fff",
                    "color": "#3c4043",
                    "fontFamily": "Roboto,Arial,sans-serif",
                    "fontSize": "12px",
                    "left": "10px",
                    "position": "absolute",
                    "textAlign": "center",
                    "textTransform": "uppercase",
                    "top": "-6px",
                    "width": "32px",
                  }
                }
              >
                hsl
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`GoogleFields renders correctly 1`] = `
<div
  className="flexbox-fix"
  style={
    Object {
      "display": "flex",
      "height": "100px",
      "marginTop": "4px",
    }
  }
>
  <div
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "padding": "0px 4.4px",
        }
      }
    >
      <div
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <input
          id="rc-editable-input-9"
          onBlur={[Function]}
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder={undefined}
          spellCheck="false"
          style={
            Object {
              "MozBorderRadius": "5px",
              "OBorderRadius": "5px",
              "WebkitBorderRadius": "5px",
              "border": "1px solid #dadce0",
              "borderRadius": "5px",
              "boxSizing": "border-box",
              "fontFamily": "Roboto,Arial,sans-serif",
              "fontSize": "11px",
              "height": "38px",
              "msBorderRadius": "5px",
              "outline": "none",
              "padding": "4px 10% 3px",
              "textAlign": "center",
              "textTransform": "lowercase",
              "width": "100%",
            }
          }
          value="#FF0000"
        />
        <label
          htmlFor="rc-editable-input-9"
          onMouseDown={[Function]}
          style={
            Object {
              "background": "#fff",
              "color": "#3c4043",
              "fontFamily": "Roboto,Arial,sans-serif",
              "fontSize": "12px",
              "left": "0",
              "marginLeft": "auto",
              "marginRight": "auto",
              "position": "absolute",
              "right": "0",
              "textAlign": "center",
              "textTransform": "uppercase",
              "top": "-6px",
              "width": "35px",
            }
          }
        >
          hex
        </label>
      </div>
    </div>
    <div
      style={
        Object {
          "WebkitJustifyContent": "space-between",
          "display": "flex",
          "justifyContent": "space-between",
          "paddingTop": "10px",
        }
      }
    >
      <div
        style={
          Object {
            "flexGrow": "1",
            "margin": "0px 4.4px",
          }
        }
      >
        <div
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <input
            id="rc-editable-input-10"
            onBlur={[Function]}
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder={undefined}
            spellCheck="false"
            style={
              Object {
                "MozBorderRadius": "5px",
                "OBorderRadius": "5px",
                "WebkitBorderRadius": "5px",
                "border": "1px solid #dadce0",
                "borderRadius": "5px",
                "boxSizing": "border-box",
                "fontFamily": "Roboto,Arial,sans-serif",
                "fontSize": "11px",
                "height": "38px",
                "msBorderRadius": "5px",
                "outline": "none",
                "paddingLeft": "10px",
                "textTransform": "lowercase",
                "width": "100%",
              }
            }
            value="255, 0, 0"
          />
          <label
            htmlFor="rc-editable-input-10"
            onMouseDown={[Function]}
            style={
              Object {
                "background": "#fff",
                "color": "#3c4043",
                "fontFamily": "Roboto,Arial,sans-serif",
                "fontSize": "12px",
                "left": "10px",
                "position": "absolute",
                "textAlign": "center",
                "textTransform": "uppercase",
                "top": "-6px",
                "width": "32px",
              }
            }
          >
            rgb
          </label>
        </div>
      </div>
      <div
        style={
          Object {
            "flexGrow": "1",
            "margin": "0px 4.4px",
          }
        }
      >
        <div
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <input
            id="rc-editable-input-11"
            onBlur={[Function]}
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder={undefined}
            spellCheck="false"
            style={
              Object {
                "MozBorderRadius": "5px",
                "OBorderRadius": "5px",
                "WebkitBorderRadius": "5px",
                "border": "1px solid #dadce0",
                "borderRadius": "5px",
                "boxSizing": "border-box",
                "fontFamily": "Roboto,Arial,sans-serif",
                "fontSize": "11px",
                "height": "38px",
                "msBorderRadius": "5px",
                "outline": "none",
                "paddingLeft": "10px",
                "textTransform": "lowercase",
                "width": "100%",
              }
            }
            value="0°, 100%, 100%"
          />
          <label
            htmlFor="rc-editable-input-11"
            onMouseDown={[Function]}
            style={
              Object {
                "background": "#fff",
                "color": "#3c4043",
                "fontFamily": "Roboto,Arial,sans-serif",
                "fontSize": "12px",
                "left": "10px",
                "position": "absolute",
                "textAlign": "center",
                "textTransform": "uppercase",
                "top": "-6px",
                "width": "32px",
              }
            }
          >
            hsv
          </label>
        </div>
      </div>
      <div
        style={
          Object {
            "flexGrow": "1",
            "margin": "0px 4.4px",
          }
        }
      >
        <div
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <input
            id="rc-editable-input-12"
            onBlur={[Function]}
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder={undefined}
            spellCheck="false"
            style={
              Object {
                "MozBorderRadius": "5px",
                "OBorderRadius": "5px",
                "WebkitBorderRadius": "5px",
                "border": "1px solid #dadce0",
                "borderRadius": "5px",
                "boxSizing": "border-box",
                "fontFamily": "Roboto,Arial,sans-serif",
                "fontSize": "11px",
                "height": "38px",
                "msBorderRadius": "5px",
                "outline": "none",
                "paddingLeft": "10px",
                "textTransform": "lowercase",
                "width": "100%",
              }
            }
            value="0°, 100%, 50%"
          />
          <label
            htmlFor="rc-editable-input-12"
            onMouseDown={[Function]}
            style={
              Object {
                "background": "#fff",
                "color": "#3c4043",
                "fontFamily": "Roboto,Arial,sans-serif",
                "fontSize": "12px",
                "left": "10px",
                "position": "absolute",
                "textAlign": "center",
                "textTransform": "uppercase",
                "top": "-6px",
                "width": "32px",
              }
            }
          >
            hsl
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`GooglePointer renders correctly 1`] = `
<div
  style={
    Object {
      "MozBorderRadius": "22px",
      "MozTransform": "translate(-10px, -7px)",
      "OBorderRadius": "22px",
      "OTransform": "translate(-10px, -7px)",
      "WebkitBorderRadius": "22px",
      "WebkitTransform": "translate(-10px, -7px)",
      "background": "hsl(250, 100%, 50%)",
      "border": "2px white solid",
      "borderRadius": "22px",
      "height": "20px",
      "msBorderRadius": "22px",
      "msTransform": "translate(-10px, -7px)",
      "transform": "translate(-10px, -7px)",
      "width": "20px",
    }
  }
/>
`;

exports[`GooglePointerCircle renders correctly 1`] = `
<div
  style={
    Object {
      "MozBorderRadius": "22px",
      "MozTransform": "translate(-12px, -13px)",
      "OBorderRadius": "22px",
      "OTransform": "translate(-12px, -13px)",
      "WebkitBorderRadius": "22px",
      "WebkitTransform": "translate(-12px, -13px)",
      "background": "hsl(250, 50%, 20%)",
      "border": "2px #fff solid",
      "borderRadius": "22px",
      "height": "20px",
      "msBorderRadius": "22px",
      "msTransform": "translate(-12px, -13px)",
      "transform": "translate(-12px, -13px)",
      "width": "20px",
    }
  }
/>
`;
